ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java

```diff
@@ -174,8 +176,13 @@ private void fetchRequests(RequestRegistrationCallback<DeleteOperation> requestR
    * @return The DeleteRequest.
    */
   private DeleteRequest createDeleteRequest() {
-    return new DeleteRequest(NonBlockingRouter.correlationIdGenerator.incrementAndGet(), routerConfig.routerHostname,
-        blobId, deletionTimeMs);
+    if (!enableForceDeleteInRequest) {
+      return new DeleteRequest(NonBlockingRouter.correlationIdGenerator.incrementAndGet(), routerConfig.routerHostname,
+          blobId, deletionTimeMs);
+    } else {
+      return new DeleteRequest(NonBlockingRouter.correlationIdGenerator.incrementAndGet(), routerConfig.routerHostname,
+          blobId, deletionTimeMs, DeleteRequest.DELETE_REQUEST_VERSION_3, enableForceDeleteInRequest);
```

Comment:
nit: If we come here, `enableForceDeleteInRequest` in `DeleteRequest` constructor can be replaced  `true` :)

Also, when server change is rolled out, we could also update `DeleteRequest.CURRENT_VERSION` to 3 and avoid passing version in the constructor call? 
```
return new DeleteRequest(NonBlockingRouter.correlationIdGenerator.incrementAndGet(), routerConfig.routerHostname,
          blobId, deletionTimeMs, enableForceDeleteInRequest)
```

Reply:
We can do that later.